tap_iterable/iterable.py

```diff
@@ -138,11 +142,19 @@ def templates(self, column_name, bookmark):
       "InApp",
       "SMS"
     ]
+    # `templates` API bug where it doesn't extract the records 
+    #  where `startDateTime`= 2023-03-01+07%3A31%3A15 though record exists
+    #  hence, substracting one second so that we could extract atleast one record 
+    bookmark_val = utils.strptime_with_tz(bookmark) - timedelta(seconds=1)
+    bookmark = utils.strftime(bookmark_val)
     for template_type in template_types:
       for medium in message_mediums:
-        res = self.get("templates", templateTypes=template_type, messageMedium=medium)
-        for t in res["templates"]:
-          yield t
+        for kwargs in self.get_start_end_date(bookmark):
+          res = self.get("templates", templateTypes=template_type, messageMedium=medium, **kwargs)
+          for t in res["templates"]:
+            rec_date_time = utils.strptime_with_tz(helper.epoch_to_datetime_string(t["updatedAt"]))
+            if rec_date_time >= bookmark_val:
+              yield t
```

Comment:
remove single letter variables
```suggestion
          for template in res["templates"]:
            rec_date_time = utils.strptime_with_tz(epoch_to_datetime_string(template["updatedAt"]))
            if rec_date_time >= bookmark_val:
              yield template
```

Reply:
This will be the part of refactoring code and this branch is specific to functionality changes. And, the line you have mentioned, I haven't made any changes to it.If I change this now, I have to be consistent about it for all the streams mentioned in the iterable.py.
Created the backlog ticket [TDL-22279](https://jira.talendforge.org/browse/TDL-22279) for the same and we can take care of it as part of later release as refactoring the code is not priority for the beta release.